web_scrapper.py

```diff
@@ -0,0 +1,48 @@
+import requests
+from bs4 import BeautifulSoup
+import json
+
+
+def remove_html_tags(text):
+    """Remove html tags from a string"""
+    import re
+    clean = re.compile('<.*?>')
+    return re.sub(clean, '', text)
+
+
+listing_html = requests.get(
+    "https://www.linkedin.com/jobs/software-engineer-intern-jobs-united-states/")
+
+doc = BeautifulSoup(listing_html.text, "html.parser")
+
+
+jobs_ul = doc.find("ul", {"class": "jobs-search__results-list"})
+
+data_dump = {}
+for li in jobs_ul.find_all("li"):
+    try:
+        posting_url = li.div.a['href']
+        posting_url = posting_url.split("?", 1)[0]
+        temp_dict = {}
+        posting_html = requests.get(
+            posting_url)
+        listing = BeautifulSoup(posting_html.text, "html.parser")
+        job_title = listing.find(
+            "h1", {"class": "top-card-layout__title topcard__title"}).string
+        temp_dict['job_title'] = job_title.strip()
+        company_name = listing.find(
+            "a", {"class": "topcard__org-name-link"}).string
+        temp_dict['company_name'] = company_name.strip()
+        location = listing.find(
+            "span", {"class": "topcard__flavor topcard__flavor--bullet"}).string
+        temp_dict['location'] = location.strip()
+        description = listing.find(
+            "div", {"class": "description__text"}).find("div", {"class": "show-more-less-html__markup"})
+        temp_dict['description'] = remove_html_tags(str(description)).strip()
+        data_dump[posting_url] = temp_dict
+    except Exception:
+        with open('error.txt', 'w') as file:
+            file.write(posting_url)
```

Comment:
This is a good example.  Firstly, if I were to try to debug this code from the `error.txt` log file, I wouldn't know what the error was.  I would just see a URL.  Is the error that we were unable to open the job posting URL?  Secondly, this code is neither commented nor self documenting.  Below is how I would go about fixing that.

Presumably, you would first add a code comment here, something like this:
```
except Exception:
    # Save to our error log the job-posting URL that we were unable to open
    with open('error.txt', 'w') as file:
         file.write(posting_url)
```
I am not totally sure that the error is that we were unable to open the job-posting URL, but let's assume.

You would then convert this code comment into code that is more self documenting, like this:
```
except Exception:
    log_error("Unable to open job-posting URL " + job_posting_url)

def log_error(message):
    with open('error.txt', 'w') as file:
         file.write(message)
```

(The idea with self-documenting code over comments is that code evolves while comments go stale.)